src/common/utils.c

```diff
@@ -1681,3 +1689,78 @@ struct timespec timespec_abs_diff(struct timespec t1, struct timespec t2)
 	res.tv_nsec = diff % (uint64_t) NSEC_PER_SEC;
 	return res;
 }
+
+static
+size_t read_proc_meminfo_field(const char *field)
+{
+	size_t ret, value_kb, value;
+	char *name = NULL;
+	FILE *proc_meminfo;
+
+	/* Allocate the buffer to read the content of /proc/meminfo. */
+	name = zmalloc(PROC_MEMINFO_FIELD_MAX_NAME_LEN * sizeof(char));
+	if (!name) {
+		value = -1;
+		goto alloc_error;
+	}
+
+	proc_meminfo = fopen(PROC_MEMINFO_PATH, "r");
+	if (!proc_meminfo) {
+		PERROR("Failed to fopen /proc/meminfo");
+		value = -1;
```

Comment:
`value` is an unsigned value (`size_t`).

Reply:
hum should this function return a `ssize_t` in order to return an error?